### PR TITLE
[FrameworkBundle] Add ability to override name of the config directory

### PR DESCRIPTION
**Pull request by ivanstan**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | yes 
| Deprecations? | no 
| Tickets       | Fix #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

For way too long Symfony is following Linux directory structure for all of its directories except for config, if we don't plan to change `/config` to `/etc` at least we can avoid hard-coding it and allow overriding of the method `getConfigDir` in Kernel.php

### CODE REVIEW

**Comment by sstok** [comment]
Please use a descriptive title.

FYI. This will require an update to Flex-plugin as well as recipes expect this convention.

**Comment by stof** [comment]
> FYI. This will require an update to Flex-plugin as well as recipes expect this convention.

If we don't change the default, I don't think it will impact Flex (other parts of Flex already assume you use the standard symfony structure rather than a customized one)

**Comment by ivanstan** [comment]
> > FYI. This will require an update to Flex-plugin as well as recipes expect this convention.
> 
> If we don't change the default, I don't think it will impact Flex (other parts of Flex already assume you use the standard symfony structure rather than a customized one)

I think preferred approach would be to first implement a function that returns a config directory path without changing the default, with a possibility for a user to override it on his own responsibility. This will give some time to the flex and recipe developers to start relying on this function before decision is made if actually the directory name is gonna change.

**Comment by Tobion** [comment]
@sstok probably means it somewhat requires an update of the recipes, e.g. https://github.com/symfony/recipes/blob/master/symfony/framework-bundle/5.4/src/Kernel.php to make use of the config dir method.

I'm also 👎 as there are is not enough demand for such a change. People who really want to diverge from documentation and flex support etc, can just overwrite the relevant kernel methods themselves.

**Comment by Nyholm** [comment]
Hm.. I agree with @Tobion that it is not a great demad for this. 

However, this patch is super small and Im not sure we have any other hard coded path like this. 

This will (or SHOULD) obviously only be used by super advanced that have really good reasons by doing so. Just "I want to follow Unix standards" is not really a strong argument. 

So, Yes, (almost) nobody should use this. But Im fine with this small patch anyways. 

@ivanstan, may I ask you if you have a good reason for using `/etc` instead? What part of your work/application would be simpler if you could use `/etc`

**Comment by ivanstan** [comment]
@Nyholm To be honest its just my OCD. Other than the fact that all other directories are configurable.

**Comment by Nyholm** [comment]
Hehe. I’ve been there. But I guess that is something you need to fight though or override registerContainerConfiguration() and registerBundles() yourself. 

I’m leaning towards closing this. I’ll wait an opinion from some other maintainer.

**Comment by ivanstan** [comment]
@Nyholm Yes, I have those methods overridden in most of my projects :)

**Review by nicolas-grekas — COMMENTED** [review]
I would be +0 if the method were private.

**Comment by ro0NL** [comment]
im +1 if we update https://github.com/symfony/recipes/blob/master/symfony/framework-bundle/5.4/src/Kernel.php :)

File: src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php [hunk 1]
```diff
@@ -124,7 +129,7 @@ public function registerContainerConfiguration(LoaderInterface $loader)
             $kernelDefinition->addTag('routing.route_loader');
 
             $container->addObjectResource($this);
-            $container->fileExists($this->getProjectDir().'/config/bundles.php');
```
Comment (yceruto) [inline src/Symfony/Bundle/FrameworkBundle/Kernel/MicroKernelTrait.php:127]: There are other occurrences in `FrameworkBundle/DependencyInjection/FrameworkExtension.php` ( `<project_dir>/config/validator` and `<project_dir>/config/serializer` ).

Also, we are hardcoding it here:
https://github.com/symfony/symfony/blob/46b2be98e88a05191181199bd514263210b7ab03/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php#L163

**Review by yceruto — COMMENTED** [review]
I'm also not sure if it will be worth it, from what I've read in other comments, and because `config/` dir matches package standards (https://github.com/php-pds/skeleton).

If we do it for apps, why don't we do the same for bundles?

I'm -1 so far.

**Comment by fabpot** [comment]
Done as part of #42991. Thank you.

**Pull request closed without merging**